### DISCUSSION TITLE
How to use hierarchy in threads panel of debugger from the debug adapter's view?

### DISCUSSION THREAD

**Discussion by EvilBeaver** [eclipse-theia/theia#12106] [Q&A] [answered]
Hello! I'm implementing a debug adapter with DAP protocol. And I see that Threads panel in Theia supports hierarchy.

Question is: how can I use this feature from debug adapter, i.e. create an item "debug session" and two child items "server thread" and "client thread"? 

It seems Debug Adapter Protocol tells nothing about thread hierarchy. How to use this hierarchical threads panel? What it's child-parent relationships means?

**Comment by msujew** [comment, marked as answer]
Hi @EvilBeaver,

Debug adapters have little say in what populates the threads panel, but rather vscode's [debug sessions](https://code.visualstudio.com/api/references/vscode-api#DebugSession). Each thread is usually implemented as its own debug session with the main session as the `parentSession`.

**Reply by EvilBeaver** [reply]
So how can I manage child items in this panel? If not from debug adapter - then it should be my own plugin or what? Where should I write some extension points for this task?
